src/org/opensolaris/opengrok/authorization/AuthorizationFramework.java

```diff
@@ -683,31 +762,52 @@ private boolean checkAll(HttpServletRequest request, String cache, Nameable enti
 
         stats.addRequest("authorization_cache_misses");
 
-        long time = System.currentTimeMillis();
-
-        boolean overallDecision = performCheck(entity, pluginPredicate, skippingPredicate);
+        long time = 0;
+        boolean overallDecision = false;
+        
+        lock.readLock().lock();
+        try {
+            // Make sure there is a HTTP session that corresponds to current plugin version.
+            HttpSession session;
+            if (((session = request.getSession(false)) != null) && isSessionInvalid(session)) {
```

Comment:
Can you please check if this works with the JSONServlet?

From now we kinda assume that plugins use http session. But I guess this is the common use case.

Reply:
Checked the servlet with `http://localhost:8084/source/json?freetext=foo&maxresults=80`, returns valid JSON.

Reply:
perfect